Capabilities api

danil-topc opened the pull request:
Add class with capabilities api endpoint
Based on #11, will rebase to master after #11 will be merged

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/8c1f5f9e-7d86-4b7c-b2a2-ae3d70585297) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/8c1f5f9e-7d86-4b7c-b2a2-ae3d70585297) completed: **12 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/b31f511f-a22d-420b-bcd5-3e457074524d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/b31f511f-a22d-420b-bcd5-3e457074524d) completed: **1 new issues, 3 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/3ce093f3-9656-4d3c-a6b1-8b8a6d8df897) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/9dcd9727-b69b-43fe-aa8a-ff6ad07bbdfb) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/9dcd9727-b69b-43fe-aa8a-ff6ad07bbdfb) completed: **3 updated code elements**

matejak commented on tests/test_capabilities.py:
```diff
@@ -0,0 +1,8 @@
+
+    def test_get_capabilities(self):
+        res = self.nxc.get_capabilities()
+        assert res['ocs']['meta']['statuscode'] == self.SUCCESS_CODE
```
Great work, I propose one modification - don't check for the whole version, but go only for the major version (i.e. the `X` in v. `X.Y.Z`). Major versions don't change so often, [they are part of the image name](https://hub.docker.com/_/nextcloud/?tab=tags&page=3), and they test the capability query anyway. On recent-enough Docker setups, you [can parametrize the FROM command](https://stackoverflow.com/questions/47156357/dockerfile-parameterized-from) using an argument that you can propagate to an environmental variable, so you don't have to repeat the literal nextcloud version on multiple places.

danil-topc replied:
Done, moved nextcloud version to parameter and check only major version in test.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/dd41283c-b228-4421-b438-91c24050aacb) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/dd41283c-b228-4421-b438-91c24050aacb) completed: **3 updated code elements**

matejak commented:
Hello Danil, I generally like your PR, but it needs some ironing out. If you remove all containers, pull & rebuild, it is likely that tests won't pass due to nextcloud app not initialized properly. I suggest to throw away postgres in favor to sqlite as outlined in #12, it helped me and also solves the problem with the postgres persistence side-effect.
Next, you have spurious `NEXTCLOUD_USERNAME=admin` and `NEXTCLOUD_PASSWORD=admin` that aren't used elsewhere in the `.env` file,
Aside from that, the PR is good for merge!

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/90978152-3617-430f-9d0f-d66bb8359c5c) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/90978152-3617-430f-9d0f-d66bb8359c5c) completed: **1 new issues, 3 updated code elements**

danil-topc commented:
Hello Matej, thanks for review! I switched from postgres to sqlite as in #12 and deleted unused variables.
Also, after changing to sqlite in groupfolders app tests, nextcloud started to return strings instead of integers as before. Didn't investigate it much, just changed transforming both values to strings.

matejak commented:
It is great that we discovered ourselves that those data types may be somehow arbitrary and that we have to pay attention to them.
Aside from this, the PR looks good. I will merge the LDAP one first, then please rebase this one, and I will merge it.

danil-topc commented:
I got rid of the aliases, thanks, and will rebase the branch after #17 merge.

matejak commented:
Great, I appreciate that. I see that you took the `8080` port mapping out too, I agree with that, but I also learned that it may come in handy when debugging e.g. installation issues. So if you later want to add it back, don't hesitate.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/21648e80-5ff7-416c-acb2-8e577167d617) was created.

danil-topc commented:
Yeah, I just left ports uncommitted on local, but it should probably be in repository too. Returned it and rebased.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/EnterpriseyIntranet/nextcloud-API/inspections/21648e80-5ff7-416c-acb2-8e577167d617) completed: **3 updated code elements**

matejak commented:
Good job!
